src/boxplot.jl

```diff
@@ -118,3 +118,47 @@ notch_width(q2, q4, N) = 1.58 * (q4-q2)/sqrt(N)
     ()
 end
 Plots.@deps boxplot shape scatter
+
+
+# ------------------------------------------------------------------------------
+# Grouped Boxplot
+
+@userplot GroupedBoxplot
+
+recipetype(::Val{:groupedboxplot}, args...) = GroupedBoxplot(args)
+
+@recipe function f(g::GroupedBoxplot; spacing = 0.1)
+    x, y = grouped_xy(g.args...)
+
+    # extract xnums and set default bar width.
+    # might need to set xticks as well
+    x = if eltype(x) <: Number
+		bar_width --> (0.8 * mean(diff(x)))
+        float.(x)
+    else
+		bar_width --> 0.8
+        ux = unique(x)
```

Comment:
I'm a bit unsure what's going on here in terms of when the "grouping" is done. How can this sidestep the machinery in https://github.com/JuliaPlots/Plots.jl/blob/399a9b16f6b4a9259c5c844b38954bc56715a1ba/src/args.jl#L1062 and https://github.com/JuliaPlots/Plots.jl/blob/42aa7d8d55e872c105a45c70dc7b2e9bef6f00e6/src/series.jl#L634?

Reply:
I think the user recipe is applied before the grouping recipe, no?